### PR TITLE
Updated with Linear Space Solution

### PR DESCRIPTION
**Pull request by shoaibrayeen**
Subset sum problem takes O(n^2) space for the table. We can reduce the space to linear time.
Here's how I reduced the space.
The function my_submission() contains the solution with the linear space.
You can run it on the local machine.
Thank you!

### CODE REVIEW

File: dynamic_programming/sum_of_subset.py [hunk 1]
```diff
@@ -25,10 +26,40 @@ def isSumSubset(arr, arrLen, requiredSum):
...
+                subset[i][j] = (subset[(i+1)%2][j] or subset[(i+1)%2][j-arr[i-1]])
+
+ 
+    return subset[arrLen%2][requiredSum]
+
+
 arr = [2, 4, 6, 8]
 requiredSum =  5
 arrLen = len(arr)
 if isSumSubset(arr, arrLen, requiredSum):
```
Comment (harshildarji) [inline dynamic_programming/sum_of_subset.py]: You also changed the method name above, but forgot to update it here! So it throws an error: `undefined name 'isSumSubset'`

File: dynamic_programming/sum_of_subset.py [hunk 2]
```diff
@@ -25,10 +26,40 @@ def isSumSubset(arr, arrLen, requiredSum):
...
+    #sum is not zero and set is empty then false
+    for i in range(1, requiredSum + 1):
+        subset[0][i] = False
+
+    for i in range(1, arrLen + 1):
+        for j in range(1, requiredSum + 1):
+            if arr[i-1] > j:
+                subset[i % 2][j] = subset[(i+1)%2][j]
+            if arr[i-1]<=j:
+                subset[i][j] = (subset[(i+1)%2][j] or subset[(i+1)%2][j-arr[i-1]])
```
[the remark this answers is not in the document]
Comment (shoaibrayeen) [reply dynamic_programming/sum_of_subset.py]: The program works fine now. You could check it.

**Comment by shoaibrayeen** [comment]
Now, it works fine and updated with linear space solution.

**Comment by shoaibrayeen** [comment]
> Now, it works fine and updated with linear space solution.

**Comment by shoaibrayeen** [comment]
Now, it works fine and updated with linear space solution. Why can't the pull request merge?

**Comment by cclauss** [comment]
This PR could benefit from tests https://travis-ci.org/TheAlgorithms/Python/builds/585697557#L640 and type hints https://docs.python.org/3/library/typing.html

**Pull request closed without merging**